src/Exchange/exg-gwfgwe.f90

```diff
@@ -220,6 +228,34 @@ subroutine exg_ar(this)
       call store_error(errmsg, terminate=.TRUE.)
     end if
     !
+    ! -- Make sure idomains are identical
+    select type (gwfdis => gwfmodel%dis)
```

Comment:
All the `select type` statements are unfortunate, but idomain is defined separately for each dis type since it's the real grid shape instead of flat

Reply:
Ahh, bummer.  I guess now that all discretizations have IDOMAIN, we could move it to disbasetype and make it a 1D int vector, but that is for another time, I think.

Reply:
If I am understanding correctly, we are trying to assess the equality of grid. Maybe we can have a `function is_equal(this, other_dis) result(are_equal)` to delegate the check to the particular Discretization module?

Reply:
PS: happy to help and chat more about that option

Reply:
@mjr-deltares thanks we should definitely chat on this with @langevin-usgs 

It may be relevant to #2153 as well. In that PR, I first planned to load the full flow model dis into fmi, then an `is_equal` function would be quite natural to use. But the dis types have some strong expectations about how they are initialized and where they get their data from. So in the near term, I was thinking of loading only the minimum necessary info to do the same checks. Hopefully we can come up with a nicer long term solution.

Reply:
This is a good point.  In this new check, we are saying the grids are the same if the number of nodes and idomain is the same; it is easy to accidentally leave out idomain when using flopy to construct dis.  But certainly, if tops and bottoms were also different, then that would not be detected here, and problems would be very difficult to find.  A discretization-based `is_equal()` (and maybe `is_same()`?) seems like a nice way to ensure grid equality.

Reply:
OK, let's talk tomorrow and plan a meeting. Btw I had no intention stalling this PR. This should probably go in and then we refactor when we have a plan.